TESTS/mbedmicro-rtos-mbed/rtostimer/main.cpp

```diff
@@ -36,6 +36,19 @@ void timer_callback(void const *arg)
     sem->release();
 }
 
+/* In order to successfully run this test suite when compiled with --profile=debug
+ * error() has to be redefined as noop.
+ *
+ * RtosTimer calls RTX API which uses Event Recorder functionality. When compiled
+ * with MBED_TRAP_ERRORS_ENABLED=1 (set in debug profile) EvrRtxTimerError() calls error()
+ * which aborts test program.
+ */
+#if defined(MBED_TRAP_ERRORS_ENABLED) && MBED_TRAP_ERRORS_ENABLED
+void error(const char* format, ...) {
```

Comment:
Should we validate in the function that causes an error it actually happens by setting some flag here?

Reply:
Isn't that going too deep into internals? The `RtosTimer` API does not say anything about calling `error()`.

Reply:
True, you're right.